qa/os/src/test/java/org/elasticsearch/packaging/test/PasswordToolsTests.java

```diff
@@ -0,0 +1,140 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.packaging.test;
+
+import org.apache.http.client.fluent.Request;
+import org.elasticsearch.packaging.util.Distribution;
+import org.elasticsearch.packaging.util.FileUtils;
+import org.elasticsearch.packaging.util.ServerUtils;
+import org.elasticsearch.packaging.util.Shell;
+import org.junit.Before;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.packaging.util.FileUtils.append;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.collection.IsMapContaining.hasKey;
+import static org.junit.Assume.assumeTrue;
+
+public class PasswordToolsTests extends PackagingTestCase {
+
+    private static final Pattern USERPASS_REGEX = Pattern.compile("PASSWORD (\\w+) = ([^\\s]+)");
+    private static final String BOOTSTRAP_PASSWORD = "myS3curepass";
+
+    @Before
+    public void filterDistros() {
+        assumeTrue("only default distro", distribution.flavor == Distribution.Flavor.DEFAULT);
+        assumeTrue("no docker", distribution.packaging != Distribution.Packaging.DOCKER);
+    }
+
+    public void test010Install() throws Exception {
+        install();
+        append(installation.config("elasticsearch.yml"),
+            "xpack.license.self_generated.type: trial\n" +
+            "xpack.security.enabled: true");
+    }
+
+    public void test20GeneratePasswords() throws Exception {
+        assertWhileRunning(() -> {
+            Shell.Result result = installation.executables().elasticsearchSetupPasswords.run(sh, "auto --batch", null);
+            Map<String, String> userpasses = parseUsersAndPasswords(result.stdout);
+            for (Map.Entry<String, String> userpass : userpasses.entrySet()) {
+                String response = ServerUtils.makeRequest(Request.Get("http://localhost:9200"), userpass.getKey(), userpass.getValue());
+                assertThat(response, containsString("You Know, for Search"));
+            }
+        });
+    }
+
+    public void test30AddBootstrapPassword() throws Exception {
+
+        try (Stream<Path> dataFiles = Files.list(installation.data)) {
+            // delete each dir under data, not data itself
+            dataFiles.forEach(file -> {
+                if (distribution.platform != Distribution.Platform.WINDOWS) {
+                    FileUtils.rm(file);
+                }
```

Comment:
should we have an `else` here or a return?

Reply:
oops, yes, i meant for a return.